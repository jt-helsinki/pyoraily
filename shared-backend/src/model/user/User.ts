/*
 *
 * MIT License.
 *
 */
import { UserRole } from './UserRole';
import { UserStatus } from './UserStatus';

export enum Gender {
  MALE = 'male',
  FEMALE = 'female',
  NA = 'na',
}

export interface User {
  id: string;
  firstName: string;
  lastName: string;
  fullName?: string;
  yearOfBirth?: number;
  gender?: Gender;
  email: string;
  uciId?: number;
  status: UserStatus;
  createdAt: string;
  updatedAt?: string;
  lastLogin?: string;
  userRoles: UserRole[];
}
