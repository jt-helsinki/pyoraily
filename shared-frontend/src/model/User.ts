/*
 *
 * MIT License.
 *
 */

import { UserRole } from './UserRole';
import { UserStatus } from './UserStatus';

export interface User {
  id: string | null;
  firstName: string;
  lastName: string;
  gender: 'male' | 'female' | 'na';
  email: string;
  uciId?: string;
  yearOfBirth: number;
  status: UserStatus;
  createdAt?: string;
  updatedAt?: string;
  lastLogin?: string;
  userRoles: UserRole[];
}
