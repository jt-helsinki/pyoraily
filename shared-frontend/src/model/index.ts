/*
 *
 * Copyright (c) Vaisala Oyj. All rights reserved.
 *
 */

export * from './AthleteProfile';
export * from './Event';
export * from './Result';
export * from './types';
export * from './User';
export * from './UserPreferences';
export * from './UserRole';
export * from './UserStatus';
