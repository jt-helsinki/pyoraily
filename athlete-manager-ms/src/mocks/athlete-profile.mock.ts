/*
 *
 * MIT License.
 *
 */
import { AthleteProfile } from '@src/model/AthleteProfile';
import { Event } from '@src/model/Event';
import { Gender } from 'pyoraily-shared-backend/model/user/User';

export const mockAthleteProfile: AthleteProfile = {
  id: '123456789',
  userID: '123456789',
  year: 2023,
  firstName: 'John',
  lastName: 'Doe',
  yearOfBirth: 2000,
  gender: Gender.MALE,
  nominatedCategory: 'Junior',
  nominatedDisciplines: [] as string[],
  events: [] as Event[],
};
