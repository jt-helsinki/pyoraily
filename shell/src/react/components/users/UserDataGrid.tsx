/*
 *
 * MIT License.
 *
 */
import { dataGridSortComparator } from 'pyoraily-shared-frontend/utils/sortComparator';
import * as UserUtils from 'pyoraily-shared-frontend/utils/UserUtils';
import { User } from 'pyoraily-shared-frontend/model/User';
import { UserRole } from 'pyoraily-shared-frontend/model/UserRole';
import * as UserComponentUtils from '@src/react/components/common/UserComponentUtils';
import * as React from 'react';
import DataGrid, { CellRendererProps, Column, SortColumn } from 'react-data-grid';
import { Button, Dimmer, Loader } from 'semantic-ui-react';

const columns = (userRoles: UserRole[], onEditRecord: (user: User) => void): Column<User>[] => [
  {
    key: 'fullName',
    name: 'Name',
    sortable: true,
    resizable: true,
    formatter: ((record: CellRendererProps<User, null>): React.ReactNode => {
      const user: User = record.row;

      return user.firstName ? (
        <>
          {user.firstName} {user.lastName}
        </>
      ) : (
        <>Not entered</>
      );
    }) as any,
  },
  {
    key: 'email',
    name: 'Email',
    sortable: true,
    resizable: true,
    formatter: ((record: CellRendererProps<User, null>) => <span>{record.row.email || ''}</span>) as any,
  },
  {
    key: 'status',
    name: 'Status',
    resizable: true,
    formatter: ((record: CellRendererProps<User, null>): React.ReactNode => {
      const user: User = record.row;

      return <>{UserComponentUtils.userStatusToTags(user.status)}</>;
    }) as any,
  },
  {
    key: 'roles',
    name: 'Role(s)',
    sortable: false,
    resizable: true,
    formatter: ((record: CellRendererProps<User, null>): React.ReactNode => {
      const userRolesForUser = UserUtils.userRolesForUser(record.row, userRoles || []);
      return UserComponentUtils.userRolesToTags(userRolesForUser || []);
    }) as any,
  },
  {
    key: 'createdAt',
    name: 'Created',
    sortable: true,
    resizable: true,
    formatter: (record: CellRendererProps<User, null>) =>
      (<span>{new Date(record.row.createdAt as string).toLocaleDateString('en-GB')}</span>) as any,
  },
  {
    key: 'updatedAt',
    name: 'Updated',
    sortable: true,
    resizable: true,
    formatter: (record: CellRendererProps<User, null>) =>
      (<span>{new Date(record.row.updatedAt as string).toLocaleDateString('en-GB')}</span>) as any,
  },
  {
    key: 'lastLogin',
    name: 'Last Login',
    sortable: true,
    resizable: true,
    formatter: (record: CellRendererProps<User, null>) =>
      (<span>{new Date(record.row.lastLogin as string).toLocaleDateString('en-GB')}</span>) as any,
  },
  {
    key: 'edit',
    name: 'Actions',
    sortable: false,
    resizable: true,
    formatter: ((record: CellRendererProps<User, null>) => {
      const onClick = (): void => onEditRecord(record.row);
      return (
        <Button type="button" kind="ghost" size="mini" onClick={onClick}>
          Edit
        </Button>
      );
    }) as any,
  },
];

const NoRowsFallback = (props: { isFetching: boolean }): React.ReactElement =>
  props.isFetching ? (
    <div style={{ gridColumn: '1/-1' }}>Fetching users...</div>
  ) : (
    <div style={{ gridColumn: '1/-1' }}>No users found</div>
  );

interface Props {
  searchTerm: string;

  isFetching: boolean;

  userRoles: UserRole[];

  users: User[];

  onEditUser: (user: User) => void;
}

export const UserDataGrid: React.FunctionComponent<Props> = (props: Props): React.ReactElement<Props> => {
  const dataGridColumns = columns(props.userRoles || [], props.onEditUser);
  const [sortColumns, setSortColumns] = React.useState<SortColumn[]>([]);
  const sortedRows: User[] = props.users.sort(dataGridSortComparator(sortColumns));

  const filteredRows = React.useMemo(
    (): User[] =>
      props.searchTerm.length >= 3
        ? sortedRows.filter(
            (user: User) =>
              user.email?.toLocaleLowerCase().startsWith(props.searchTerm.toLocaleLowerCase()) ||
              user.lastName?.toLocaleLowerCase().startsWith(props.searchTerm.toLocaleLowerCase()) ||
              user.firstName?.toLocaleLowerCase().startsWith(props.searchTerm.toLocaleLowerCase())
          )
        : sortedRows,
    [props.searchTerm, sortedRows, sortColumns]
  );

  return (
    <>
      <Dimmer active={props.isFetching}>
        <Loader>Fetching user data...</Loader>
      </Dimmer>
      <DataGrid
        columns={dataGridColumns as any}
        rows={filteredRows}
        sortColumns={sortColumns}
        onSortColumnsChange={setSortColumns}
        defaultColumnOptions={{ width: '1fr' }}
        headerRowHeight={40}
        style={{ resize: 'both' }}
        renderers={{ noRowsFallback: <NoRowsFallback isFetching={props.isFetching} /> }}
      />
    </>
  );
};
