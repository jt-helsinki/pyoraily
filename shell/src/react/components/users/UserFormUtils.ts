/*
 *
 * MIT License.
 *
 */
import { User } from 'pyoraily-shared-frontend/model/User';
import { UserRole } from 'pyoraily-shared-frontend/model/UserRole';
import { UserStatus } from 'pyoraily-shared-frontend/model/UserStatus';
import { DropdownItemProps } from 'semantic-ui-react/dist/commonjs/modules/Dropdown/DropdownItem';

export const userRoleToMultiSelectItem = (userRole: UserRole): DropdownItemProps[] =>
  ({
    key: userRole,
    text: userRole.replace('_', ' '),
    value: userRole,
  } as any);

export const selectedUserRoleIdsToMultiSelectItems = (
  selectedUserRoleIds: string[],
  mappedUserRoles: DropdownItemProps[]
): DropdownItemProps[] => mappedUserRoles.filter((userRole) => selectedUserRoleIds.some((id) => id === userRole.id));

export const emptyUser = (): Partial<User> => ({
  id: null,
  firstName: '',
  lastName: '',
  userRoles: [],
  email: '',
  status: UserStatus.Active,
});
