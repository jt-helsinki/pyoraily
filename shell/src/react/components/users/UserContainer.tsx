/*
 *
 * MIT License.
 *
 */
import { User } from 'pyoraily-shared-frontend/model/User';
import { UserDataGrid } from '@src/react/components/users/UserDataGrid';
import * as UserFormUtils from '@src/react/components/users/UserFormUtils';
import { UserUpsertForm } from '@src/react/components/users/UserUpsertForm';
import {
  useCreateUserServerStateStore,
  useFetchUserRolesServerStateStore,
  useFetchUsersServerStateStore,
  useUpdateUserServerStateStore,
} from '@src/react/hooks/state/server/useUserServerStateStore';
import { NotificationKey } from '@src/react/types/NotificationKey';
import * as UiService from '@src/services/UiService';
import * as React from 'react';
import { Dimmer, Input, Loader, Modal } from 'semantic-ui-react';

enum ModalMode {
  None,
  EditUser,
  CreateUser,
}

export const UserContainer: React.FunctionComponent = (): React.ReactElement | null => {
  const { data: userRoles } = useFetchUserRolesServerStateStore();
  const { isFetching, data: users, refetch } = useFetchUsersServerStateStore();

  const { mutateAsync: updateUserMutate } = useUpdateUserServerStateStore();
  const { mutateAsync: createUserMutate } = useCreateUserServerStateStore();

  const [error, setError] = React.useState<string>('');
  const [isSubmitting, setIsSubmitting] = React.useState<boolean>(false);
  const [searchTerm, setSearchTerm] = React.useState<string>('');
  const [user, setUser] = React.useState<Partial<User>>(UserFormUtils.emptyUser());
  const [modalMode, setModalMode] = React.useState<ModalMode>(ModalMode.None);

  const onCancel = (): void => {
    setError(null as any);
    setIsSubmitting(false);
    setUser(UserFormUtils.emptyUser());
    setModalMode(ModalMode.None);
  };

  const handleSaveResponse = (response: Promise<User>): void => {
    setIsSubmitting(true);
    response
      .then((savedUser: User) => {
        UiService.notificationToasterSuccess(
          NotificationKey.USERS_EMAIL_UPDATED,
          `User ${savedUser.firstName} ${savedUser.lastName} <${savedUser.email}> updated.`
        );
        refetch().then(() => onCancel());
      })
      .catch((responseError: string) => {
        setError(responseError);
        setIsSubmitting(false);
      });
  };

  const onEditUser = (userToEdit: User): void => {
    setUser(userToEdit);
    setModalMode(ModalMode.EditUser);
  };

  const onEditUserSave = (userToSave: any): void => {
    handleSaveResponse(updateUserMutate(userToSave));
  };

  const visible: boolean = modalMode === ModalMode.EditUser || modalMode === ModalMode.CreateUser;

  return (
    <>
      <div>
        <p>
          <Input icon="search" placeholder="Search..." onChange={(event: any) => setSearchTerm(event.target.value)} />
        </p>
        <UserDataGrid
          searchTerm={searchTerm}
          users={users || []}
          userRoles={userRoles || []}
          isFetching={isFetching || isSubmitting}
          onEditUser={onEditUser}
        />
      </div>
      <Modal onClose={onCancel} open={visible}>
        <Dimmer active={isSubmitting}>
          <Loader>Saving</Loader>
        </Dimmer>
        {modalMode === ModalMode.EditUser ? (
          <UserUpsertForm
            isSubmitting={isSubmitting}
            title="Edit user"
            description="Change user details, permissions and roles."
            onSubmit={onEditUserSave}
            onCancel={onCancel}
            user={user as User}
            userRoles={userRoles || []}
            errorMessage={error}
          />
        ) : null}
      </Modal>
    </>
  );
};
