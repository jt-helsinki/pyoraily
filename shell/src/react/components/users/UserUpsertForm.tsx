/*
 *
 * MIT License.
 *
 */
import { UserEditFormModelSchema } from '@src/model/schema/UserFormSchemas';
import { User } from 'pyoraily-shared-frontend/model/User';
import { UserRole } from 'pyoraily-shared-frontend/model/UserRole';
import * as UserFormUtils from '@src/react/components/users/UserFormUtils';
import { Formik, FormikProps } from 'formik';
import * as React from 'react';
import { Button, Dropdown, Form, Modal, Radio } from 'semantic-ui-react';
import { DropdownItemProps } from 'semantic-ui-react/dist/commonjs/modules/Dropdown/DropdownItem';

interface Props {
  isSubmitting: boolean;

  title: string;

  description: string;

  disableNameAndEmail?: boolean;

  user: User;

  errorMessage?: string;

  onSubmit: (values: User) => void;

  onCancel: () => void;

  userRoles: UserRole[];
}

export const UserUpsertForm: React.FunctionComponent<Props> = (props: Props): React.ReactElement<Props> | null => {
  const mappedUserRoles = React.useMemo(
    (): DropdownItemProps[] => props.userRoles.map(UserFormUtils.userRoleToMultiSelectItem),
    [props.userRoles]
  );
  return (
    <>
      <Modal.Header>{props.title}</Modal.Header>
      <Modal.Content>
        <Modal.Description>{props.description}</Modal.Description>
        <Formik
          validateOnChange={true}
          validateOnBlur={true}
          enableReinitialize={true}
          initialValues={props.user}
          validationSchema={UserEditFormModelSchema}
          onSubmit={(values: User): void => {
            props.onSubmit(values);
          }}>
          {(formik: FormikProps<User>): React.ReactElement => (
            <Form id="user-upsert-form">
              <Form.Input
                type="text"
                required={true}
                id="user-upsert-form-first-name"
                name="firstName"
                label="First name"
                value={formik.values.firstName}
                disabled={props.isSubmitting}
                error={formik.errors.firstName}
                onChange={formik.handleChange}
              />
              <Form.Input
                type="text"
                required={true}
                id="user-upsert-form-last-name"
                name="lastName"
                label="Last name"
                value={formik.values.lastName}
                disabled={props.isSubmitting}
                error={formik.errors.lastName}
                onChange={formik.handleChange}
              />
              <Form.Input
                type="text"
                required={true}
                id="user-upsert-form-email"
                name="email"
                label="Email"
                value={formik.values.email}
                disabled={props.isSubmitting}
                error={formik.errors.email}
                onChange={formik.handleChange}
              />
              <Form.Field>
                <label>Roles</label>
                <Dropdown
                  id="user-upsert-form-user-roles"
                  name="userRoles"
                  placeholder="Roles"
                  fluid={true}
                  multiple={true}
                  selection
                  options={mappedUserRoles}
                  disabled={props.isSubmitting}
                  defaultValue={formik.values.userRoles}
                  error={formik.errors.userRoles as any}
                  onChange={(event: any, data: any) => formik.setFieldValue('userRoles', data.value, false)}
                />
              </Form.Field>
              <Form.Group inline={true} unstackable={false}>
                <label>Status</label>
                <Form.Field
                  control={Radio}
                  label="Active"
                  value="Active"
                  name="status"
                  disabled={props.isSubmitting}
                  checked={formik.values.status === 'Active'}
                  onClick={() => formik.setFieldValue('status', 'Active', false)}
                />
                <Form.Field
                  control={Radio}
                  label="Blocked"
                  value="Blocked"
                  name="status"
                  disabled={props.isSubmitting}
                  checked={formik.values.status === 'Blocked'}
                  onClick={() => formik.setFieldValue('status', 'Blocked', false)}
                />
              </Form.Group>
              <Button secondary={true} onClick={props.onCancel}>
                Cancel
              </Button>
              <Button primary={true} onClick={formik.submitForm}>
                Save
              </Button>
            </Form>
          )}
        </Formik>
      </Modal.Content>
    </>
  );
};

UserUpsertForm.defaultProps = {
  disableNameAndEmail: false,
};
