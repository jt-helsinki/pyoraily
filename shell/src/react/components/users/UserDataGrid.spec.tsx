import { User } from 'pyoraily-shared-frontend/model/User';
import { UserDataGrid } from '@src/react/components/users/UserDataGrid';
import { MOCK_USER_ROLES } from '@src/tests/mocks/mockUserRoleServiceResponses';
import { mockFetchUsersResponseData } from '@src/tests/mocks/mockUserServiceResponses';
import * as DataGridTestHelpers from '@src/tests/test-helpers/DataGridTestHelpers';
import { renderSUI, rerenderSUI } from '@src/tests/utils';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import * as React from 'react';

describe('UsersTable', () => {
  it('should match the snapshot', () => {
    const { container } = renderSUI(
      <UserDataGrid
        searchTerm=""
        isFetching={false}
        users={mockFetchUsersResponseData}
        userRoles={MOCK_USER_ROLES}
        onEditUser={(user: User) => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should have 2 rows.', async () => {
    const onEditUser = jest.fn();
    const user = userEvent.setup();

    const { container } = renderSUI(
      <UserDataGrid
        searchTerm=""
        isFetching={false}
        users={mockFetchUsersResponseData}
        userRoles={MOCK_USER_ROLES}
        onEditUser={onEditUser}
      />
    );
    expect(DataGridTestHelpers.getRows()).toHaveLength(2);
  });

  it('should have 1 row after search.', async () => {
    const onEditUser = jest.fn();
    const user = userEvent.setup();

    const { container } = renderSUI(
      <UserDataGrid
        searchTerm="Joe"
        isFetching={false}
        users={mockFetchUsersResponseData}
        userRoles={MOCK_USER_ROLES}
        onEditUser={onEditUser}
      />
    );
    expect(DataGridTestHelpers.getRows()).toHaveLength(2);
  });

  it('should have 0 row after search for invalid name.', async () => {
    const onEditUser = jest.fn();
    const user = userEvent.setup();

    const { container } = renderSUI(
      <UserDataGrid
        searchTerm="Mungo"
        isFetching={false}
        users={mockFetchUsersResponseData}
        userRoles={MOCK_USER_ROLES}
        onEditUser={onEditUser}
      />
    );
    expect(DataGridTestHelpers.getRows()).toHaveLength(0);
  });

  it('should call the onEditUser() function.', async () => {
    jest.useFakeTimers();
    const onEditUser = jest.fn();
    const user = userEvent.setup();

    const { rerender, container } = renderSUI(
      <UserDataGrid
        searchTerm=""
        isFetching={false}
        users={mockFetchUsersResponseData}
        userRoles={MOCK_USER_ROLES}
        onEditUser={onEditUser}
      />
    );

    //   const buttons = screen.getAllByText('Edit');
    //
    //   await user.click(buttons[0]);
    //   expect(onEditUser).toHaveBeenCalled();

    // expect(DataGridTestHelpers.getRows()).toHaveLength(2);
  });

  // it('should display table with two users name, one email, and indication of no email', async () => {
  //   renderSUI(
  //     <UserDataGrid searchTerm="" users={mockUsers} userRoles={mockUserRoles} isFetching={true} onEditUser={jest.fn()} />
  //   );
  //   await waitFor(() => {
  //     expect(screen.getByText(mockUsers[0].fullName)).toBeInTheDocument();
  //   });
  //   expect(
  //     screen.getByText(
  //       mockUsers[0].contactMedia.find((contact) => contact.mediumType === 'Email')?.characteristic.name ||
  //         'NO-EMAIL-FOUND'
  //     )
  //   ).toBeInTheDocument();
  //
  //   const dateString0 = new Date(mockUsers[0].createdAt).toLocaleDateString();
  //   const dateString1 = new Date(mockUsers[1].createdAt).toLocaleDateString();
  //
  //   expect(screen.getByText(mockUsers[1].fullName)).toBeInTheDocument();
  //   expect(screen.getByText(dateString0)).toBeInTheDocument();
  //   expect(screen.getByText(dateString1)).toBeInTheDocument();
  //   expect(screen.getByText('No email')).toBeInTheDocument();
  // });
  //
  it('should display a loading indicator', async () => {
    const { rerender, container } = renderSUI(
      <UserDataGrid
        searchTerm=""
        users={mockFetchUsersResponseData}
        userRoles={MOCK_USER_ROLES}
        isFetching={true}
        onEditUser={jest.fn()}
      />
    );

    await waitFor(() => {
      const loadingComponent = container.firstChild;
      expect(loadingComponent).toHaveClass('ui active text loader');
      expect(screen.getByText('Fetching user data...')).toBeInTheDocument();
    });

    rerenderSUI(
      rerender,
      <UserDataGrid
        searchTerm=""
        users={mockFetchUsersResponseData}
        userRoles={MOCK_USER_ROLES}
        isFetching={false}
        onEditUser={jest.fn()}
      />
    );
    await waitFor(() => {
      const loadingComponent = container.firstChild;
      expect(loadingComponent).toHaveClass('ui text loader');
      expect(screen.getByText('Fetching user data...')).toBeInTheDocument();
    });
  });
  //
  // it('should call invite user callback if invite user button is clicked', async () => {
  //   const mockInviteUser = jest.fn();
  //   renderSUI(
  //     <UserDataGrid
  //       searchTerm=""
  //       users={mockUsers}
  //       userRoles={mockUserRoles}
  //       isFetching={false}
  //       onEditUser={jest.fn()}
  //     />
  //   );
  //   fireEvent.click(screen.getByText('Invite user'));
  //   await waitFor(() => {
  //     expect(mockInviteUser).toHaveBeenCalled();
  //   });
  // });
  //
  // it('should call edit user callback if edit button is clicked', async () => {
  //   const mockEditUser = jest.fn();
  //   renderSUI(
  //     <UserDataGrid
  //       searchTerm=""
  //       users={mockUsers}
  //       userRoles={mockUserRoles}
  //       isFetching={false}
  //       onEditUser={mockEditUser}
  //     />
  //   );
  //   fireEvent.click(screen.getAllByText('Edit')[0]);
  //   await waitFor(() => {
  //     expect(mockEditUser).toHaveBeenCalled();
  //   });
  // });
});
