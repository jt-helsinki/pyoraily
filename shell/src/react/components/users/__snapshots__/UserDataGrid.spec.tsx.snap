// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersTable should match the snapshot 1`] = `
<div>
  <div
    class="ui text loader"
  >
    Fetching user data...
  </div>
  <div
    aria-colcount="8"
    aria-rowcount="3"
    class="rdg r104f42s7-0-0-beta-22"
    dir="ltr"
    role="grid"
    style="resize: both; grid-template-rows: 40px repeat(2, 35px); --rdg-header-row-height: 40px; --rdg-summary-row-height: 35px; --rdg-sign: 1; grid-template-columns: 1fr 1fr 80px 80px 80px 80px 80px 80px;"
  >
    <div
      aria-rowindex="1"
      class="rdg-header-row h197vzie7-0-0-beta-22"
      role="row"
      style="--rdg-grid-row-start: 1;"
    >
      <div
        aria-colindex="1"
        aria-selected="false"
        class="rdg-cell c1wupbe7-0-0-beta-22 rdg-cell-resizable celq7o97-0-0-beta-22"
        role="columnheader"
        style="grid-column-start: 1;"
        tabindex="0"
      >
        <span
          class="rdg-header-sort-cell h1e8ezgp7-0-0-beta-22"
          tabindex="-1"
        >
          <span
            class="rdg-header-sort-name h1rrblhe7-0-0-beta-22"
          >
            Name
          </span>
          <span />
        </span>
      </div>
      <div
        aria-colindex="2"
        aria-selected="false"
        class="rdg-cell c1wupbe7-0-0-beta-22 rdg-cell-resizable celq7o97-0-0-beta-22"
        role="columnheader"
        style="grid-column-start: 2;"
        tabindex="-1"
      >
        <span
          class="rdg-header-sort-cell h1e8ezgp7-0-0-beta-22"
          tabindex="-1"
        >
          <span
            class="rdg-header-sort-name h1rrblhe7-0-0-beta-22"
          >
            Email
          </span>
          <span />
        </span>
      </div>
    </div>
    <div
      aria-rowindex="2"
      class="rdg-row r1otpg647-0-0-beta-22 rdg-row-even"
      role="row"
      style="--rdg-grid-row-start: 2; --rdg-row-height: 35px;"
    >
      <div
        aria-colindex="1"
        aria-readonly="true"
        aria-selected="false"
        class="rdg-cell c1wupbe7-0-0-beta-22"
        role="gridcell"
        style="grid-column-start: 1;"
        tabindex="-1"
      >
        Joe

        Mangle
      </div>
      <div
        aria-colindex="2"
        aria-readonly="true"
        aria-selected="false"
        class="rdg-cell c1wupbe7-0-0-beta-22"
        role="gridcell"
        style="grid-column-start: 2;"
        tabindex="-1"
      >
        <span>
          test@test.com
        </span>
      </div>
    </div>
    <div
      aria-rowindex="3"
      class="rdg-row r1otpg647-0-0-beta-22 rdg-row-odd"
      role="row"
      style="--rdg-grid-row-start: 3; --rdg-row-height: 35px;"
    >
      <div
        aria-colindex="1"
        aria-readonly="true"
        aria-selected="false"
        class="rdg-cell c1wupbe7-0-0-beta-22"
        role="gridcell"
        style="grid-column-start: 1;"
        tabindex="-1"
      >
        Joe

        Mangle
      </div>
      <div
        aria-colindex="2"
        aria-readonly="true"
        aria-selected="false"
        class="rdg-cell c1wupbe7-0-0-beta-22"
        role="gridcell"
        style="grid-column-start: 2;"
        tabindex="-1"
      >
        <span>
          test@test.com
        </span>
      </div>
    </div>
    <div
      class="m1l09lto7-0-0-beta-22"
      data-measuring-cell-key="fullName"
      style="grid-column-start: 1; min-width: 80px;"
    />
    <div
      class="m1l09lto7-0-0-beta-22"
      data-measuring-cell-key="email"
      style="grid-column-start: 2; min-width: 80px;"
    />
  </div>
</div>
`;
